client/keys/add_test.go

```diff
@@ -114,3 +116,47 @@ func Test_runAddCmdBasic(t *testing.T) {
 	mockIn.Reset("\n" + password + "\n" + "fail" + "\n")
 	require.Error(t, cmd.ExecuteContext(ctx))
 }
+
+func TestAddRecoverFileBackend(t *testing.T) {
```

Comment:
afaict, this bug is also present on master. Maybe let's rebase this PR to master, and create backports to 0.42.x?

Reply:
indeed. Thanks, I'll attempt a fix and update PR target at this time!